rllib/evaluation/sample_collector.py

```diff
@@ -0,0 +1,165 @@
+from abc import abstractmethod, ABCMeta
```

Comment:
This is the Trajectory View API, which the RolloutWorker will use to store (from the env/model) and retrieve (for action calculations/postprocessing/training) samples. As per this PR, it's only a skeleton.